kibble/datastore/film_index_datasource.go

```diff
@@ -38,8 +38,8 @@ func (ds *FilmIndexDataSource) GetEntityType() reflect.Type {
 // Iterator - return a list of all films, iteration of 1
 func (ds *FilmIndexDataSource) Iterator(ctx models.RenderContext, renderer models.Renderer) (errCount int) {
 	clonedFilms := make([]*models.Film, len(ctx.Site.Films))
-	for i, f := range ctx.Site.Films {
-		clonedFilms[i] = transformFilm(f)
+	for _, f := range ctx.Site.Films {
+		clonedFilms = append(clonedFilms, transformFilm(*f))
```

Comment:
I think because we know the length its better the way it is?

Golang reallocates memory for the array every 5 appends I think? So less efficient as it would have to grow the array say 200 times for a client with 1000 films

Reply:
I'm confused...
If the site has 200 films, you are going to `make` an array with 200 empty entries, and then append 200 more entries so it will have a final length of 400 with only half of the entries populated.
The old way makes an array with 200 empty entries, and then populates them by index so the final length is 200 with no empties.

Example:
```
	a := make([]int, 10)
	for i := 0; i < 10; i++ {
		a = append(a, 1)
	}

	fmt.Println(a) // [0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1]
	fmt.Println(len(a)) // 20
```

Reply:
Nice catch.

Reply:
Ohh! Yes super good catch there @PeculiarGoat. Glad someone noticed that.

Fixed - I've kept the array created with len(ctx.Site.Films), but now use square bracket notation to change each array element.